src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs

```diff
@@ -963,6 +978,13 @@ private ISet<string> CalculateFilesToExcludeInPack(IPackTaskRequest<IMSBuildItem
                         }
                     }
 
+                    if (packageDependency.NoWarn.Count > 0)
+                    {
+                        var nowarnProperties = new HashSet<(NuGetLogCode, NuGetFramework)>();
+                        nowarnProperties.AddRange(packageDependency.NoWarn.Select(n => (n, framework.FrameworkName)));
+                        packageBuilder.PackageSpecificNoWarnProperties[packageDependency.Name] = nowarnProperties;
```

Comment:
Wouldn't this overwrite potentially different no warns for different frameworks?>

Reply:
You're right, I changed logic to handle it.
Please check now.

Reply:
Looks ok now. 

Please make sure you have a test for the scenario.